R/get_coverage_wcs.R

```diff
@@ -46,35 +47,56 @@
 #' }
 #'
 get_coverage_wcs <- function(
-    wcs = c("dtm", "dsm", "omz", "omw"),
+    wcs = c("dtm", "dsm", "omz", "omw", "dhmv"),
     bbox,
     layername,
     resolution,
-    wcs_crs = "EPSG:4258",
+    wcs_crs = c("EPSG:4258", "EPSG:31370"),
     output_crs = "EPSG:31370",
     bbox_crs = "EPSG:31370",
     version = c("1.0.0", "2.0.1"),
     ...) {
+
   # prelim check
   version <- match.arg(version)
+  wcs <- tolower(wcs) # case insensitive wcs
   wcs <- match.arg(wcs)
   wcs_crs <- match.arg(wcs_crs)
   bbox_crs <- match.arg(bbox_crs)
 
+
+  # constrain version | wcs
+  if (wcs == "dhmv") {
+    assert_that(version == "1.0.0",
+      msg = "WCS `DHMV` is incompatible with versions other than `1.0.0`.")
+  }
+
   # set url
   wcs <- switch(wcs,
     omz = "https://geo.api.vlaanderen.be/oi-omz/wcs",
     omw = "https://geo.api.vlaanderen.be/oi-omw/wcs",
     dtm = "https://geo.api.vlaanderen.be/el-dtm/wcs",
-    dsm = "https://geo.api.vlaanderen.be/el-dsm/wcs"
+    dsm = "https://geo.api.vlaanderen.be/el-dsm/wcs",
+    dhmv = "https://geo.api.vlaanderen.be/DHMV/wcs"
+  )
+
+  # assure package availability
+  stopifnot(
+    assertthat = require("assertthat", quietly = TRUE),
+    httr = require("httr", quietly = TRUE),
+    sf = require("sf", quietly = TRUE),
+    terra = require("terra", quietly = TRUE)
   )
```

Comment:
I think these assertions are not necessary. The required functions from these packages are already imported (see roxygen `@importFrom` statements

Reply:
Great - you decide how to assure available of requirements! 
I am not at all familiar with the `roxygen` procedure yet; I'm sure there were coding clubs about it. Interested to see how it raises errors upon lack of a package. If it doesn't, or if correct handling is only guaranteed in rstudio, we might anyways opt for explicit requirement assertions.

Reply:
(I have removed the assertion block)